topik/pemrograman-kompetitif-dasar/03-struktur-data-dasar/pemrograman-kompetitif-dasar_03-struktur-data-dasar.tex

```diff
@@ -11,43 +11,265 @@
+Selain itu, berdasarkan bentuknya, linked list juga dapat terbagi menjadi 2 macam, yaitu:
+\begin{itemize}
+	\item Linear Linked List, yaitu linked list dimana \alert{tail} atau node paling terakhir nextnya dihubungkan ke \alert{NULL}. Demikian pula dalam double linked list, maka \alert{head} atau node pertama prevnya dihubungkan ke NULL.
+	\item Circular Linked List, yaitu linked list dimana tail bagian nextnya akan dihubungkan ke head. Selain itu, bagian prev pada head juga akan dihubungkan ke tail. 
+\end{itemize}
+\end{frame}
+
+\begin{frame}
+\frametitle{Keuntungan Linked List}
+Mengapa kita harus menggunakan linked list? Hal ini dikarenakan linked list memiliki keuntungan dalam menangani penghapusan suatu elemen. Sebagai contoh, misal kita akan menghapus salah satu elemen pada linked list. Maka yang perlu dilakukan hanya menghapus node tersebut. Jika dibandingkan dengan array, maka penghapusan suatu elemen harus menggeser seluruh elemen setelah atau sebelum elemen yang dihapus.
+\end{frame}
+
+\begin{frame}
+\frametitle{Kerugian Linked List}
+Namun, penggunaan linked list juga memiliki kerugian. Sebagai contoh, untuk proses membaca elemen tidak semudah dalam pembacaan elemen pada array. Untuk membaca elemen pada indeks tertentu, maka harus dilakukan iterasi mulai dari node head hingga ke node yang diinginkan. Begitu pula dengan proses pengupdatean harus dilakukan iterasi dari awal terlebih dahulu.
+\end{frame}
+
+\section{Stack}
+\frame{\sectionpage}
+
+\begin{frame}
+\frametitle{Soal Pemanasan}
+
+Misal kita mendapat soal sebagai berikut
+\begin{itemize}
+	\item Anda diberikan sebuah string misal "acaabcbcd"
+	\item Cari string "abc" dalam string tersebut. Jika ditemukan maka hapus string "abc" tersebut, lalu ulangi pencarian.
+	\item Pencarian berakhir ketika tidak terdapat string "abc" lagi. Outputkan total penghapusan yang berhasil dilakukan
+	\item Contoh, pada string "acaabcbcd" terdapat sebuah string abc, dan hapus string tersebut menjadi "acabcd". Lalu, ditemukan lagi string abc dan hapus menjadi "acd". Karena tidak ditemukan lagi string "abc", maka output 2.
 \end{itemize}
 \end{frame}
 
 \begin{frame}
-\frametitle{Tips}
-Demi keseragaman dan kemudahan, gunakan command yang sudah didefinisikan di ../config.tex:
-% Komentar: meskipun ada yang menghasilkan format yang sama, tetapi menggunakan command ini mempermudah kalau ke depannya mau mengubah format, tidak perlu find and replace 1 per 1. 
+\frametitle{Mengenal Stack}
+
+Stack dapat dimisalkan seperti tumpukan piring. Jika terdapat piring baru yang ingin dimasukkan, maka piring tersebut masuk dari paling atas. Jika sebuah piring akan diambil dari tumpukan, maka yang diambil juga piring yang paling atas.
+
+\begin{figure}
+	\centering
+	\includegraphics[width=4 cm]{asset/plates-stack.jpg}
+\end{figure}
+
+\end{frame}
+
+\begin{frame}
+\frametitle{Mengenal Stack (lanj.)}
+
+Seperti pada analogi tumpukan piring, struktur data stack memiliki sifat yang sama:
```

Comment:
Push dan pop lebih tepat disebut operasi.
Kalo mau bicara tentang sifat, berarti yg dikaitkan adalah sifatnya Last In First Out.

Sebagai masukan, bisa dibuat "... sifat yang sama, yaitu LIFO ..."
Lalu dijelaskan kalo operasi yg didukung struktur data ini adalah "push" dan "pop".